File: apps/api/app/routers/storyboards.py
from fastapi import APIRouter

from app.schemas.storyboard import StoryboardInput, StoryboardOutput
from app.services.storyboard_service import generate_storyboard


router = APIRouter(prefix="/api/storyboards", tags=["storyboards"])


@router.post("/generate", response_model=StoryboardOutput)
def generate_storyboard_endpoint(input_data: StoryboardInput) -> StoryboardOutput:
    return generate_storyboard(input_data)

File: apps/web/src/api/storyboards.ts
import type { StoryboardInput, StoryboardOutput } from "../types/storyboard";

const API_BASE_URL = "http://127.0.0.1:8000";

export async function generateStoryboard(input: StoryboardInput): Promise<StoryboardOutput> {
  const response = await fetch(`${API_BASE_URL}/api/storyboards/generate`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(input),
  });

  if (!response.ok) {
    throw new Error("Storyboard generation request failed");
  }

  return (await response.json()) as StoryboardOutput;
}
